File: 2022/js/day1.js
function part1() {
    // Read the file and convert to an array
    const fs = require('fs');
    const content = fs.readFileSync('../day1.txt', 'utf8').split('\n');

    let nums = 0;
    let largest = 0;

    for (let i = 0; i < content.length; ++i) {
        if (content[i] !== '') {
            nums += Number(content[i]);
        } else {
            if (nums > largest) {
                largest = nums;
            }
            nums = 0;
        }
    }
    return largest;
}

console.log(part1());

function part2() {
    // Read the file and convert to an array
    const fs = require('fs');
    const content = fs.readFileSync('../day1.txt', 'utf8').split('\n');

    let nums = 0;
    let groups = [];

    for (let i = 0; i < content.length; ++i) {
        if (content[i] !== '') {
            nums += Number(content[i]);
        } else {
            groups.push(nums);
            nums = 0;
        }
    }
    // sort the array in reverse order
    groups.sort();
    groups.reverse();
    return groups[0] + groups[1] + groups[2];
}

console.log(part2());

File: 2022/py/day1.py
def part1():
    with open('../day1.txt') as f:
        # get every line of the file in a list
        content = f.readlines()

        # start at beginning of list and add nums to var until '\n'
        nums = 0
        largest = 0
        for num in content:
            if num != '\n':
                nums += int(num)
            else:
                # if the current group is larger than the largest, set largest to current
                if nums > largest:
                    largest = nums
                # reset the group count
                nums = 0
    return largest

print(part1())

def part2():
    with open('../day1.txt') as f:
        # get every line of the file in a list
        content = f.readlines()

        # start at beginning of list and add nums to var until '\n'
        nums = 0
        groups = []
        for num in content:
            if num != '\n':
                nums += int(num)
            else:
                # reset the group count
                groups.append(nums)
                nums = 0

    groups.sort(reverse=True)
    return groups[0] + groups[1] + groups[2]

print(part2())

